src/zenml/config/global_config.py

```diff
@@ -669,11 +669,8 @@ def zen_store(self) -> "BaseZenStore":
         Returns:
             The current zen store.
         """
-        if not self.store:
+        if self._zen_store is None:
```

Comment:
does adding a check on `self.store` after env variables solve the issue? since it will only go to return the `get_default_store_config` if no env variables where set

Reply:
Updated now to do the right thing: use the merged configuration to initialize the store.